packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts

```diff
@@ -275,6 +294,89 @@ export class AmplifyGraphqlApi extends Construct {
     outputStorageStrategy.addBackendOutputEntry(graphqlOutputKey, output);
   }
 
+  /**
+   * Create a deployment identifier.
+   * @returns the deployment identifier or undefined.
+   */
+  private getDeploymentIdentifier(): DeploymentIdentifier | undefined {
+    const deploymentType = this.node.tryGetContext(CDKContextKey.DEPLOYMENT_TYPE);
+    if (!deploymentType) {
+      return undefined;
+    }
+
+    const sandboxDeploymentIdentifier = this.makeSandboxDeploymentIdentifier(deploymentType);
+    if (sandboxDeploymentIdentifier) {
+      return sandboxDeploymentIdentifier;
+    }
+
+    const branchDeploymentIdentifier = this.makeBranchDeploymentIdentifier(deploymentType);
+    if (branchDeploymentIdentifier) {
+      return branchDeploymentIdentifier;
+    }
+
+    console.warn(`Encountering an unknown deployment type: ${deploymentType}.\n`);
```

Comment:
nit (for future PR is fine): Change verbiage, remove redundant newline: 
```suggestion
    console.warn(`Encountered an unknown deployment type: ${deploymentType}.`);
```